src/Server.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-swoole for the canonical source repository
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-swoole/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Zend\Expressive\Swoole;
+
+use Swoole\Http\Server as SwooleHttpServer;
+
+class Server
+{
+
+    /**
+     * @var string
+     */
+    protected $host;
+
+    /**
+     * @var int
+     */
+    protected $port;
+
+    /**
+     * @var int
+     */
+    protected $mode;
+
+    /**
+     * @var int
+     */
+    protected $protocol;
+
+    /**
+     * @var array
+     */
+    protected $options = [];
+
+    /**
+     * @var SwooleHttpServer
+     */
+    protected $swooleServer;
+
+    public function __construct(string $host, int $port, int $mode, int $protocol, array $options)
+    {
+        $this->host = $host;
+        $this->port = $port;
+        $this->mode = $mode;
+        $this->protocol = $protocol;
+        $this->options = $options;
+    }
+
+    /**
+     * Create a swoole server instance
+     */
+    public function createSwooleServer(array $appendOptions = []): SwooleHttpServer
+    {
+        if (! $this->swooleServer) {
+            $server = new SwooleHttpServer($this->host, $this->port, $this->mode, $this->protocol);
+            if ($options = array_replace($this->options, $appendOptions)) {
+                $server->set($options);
+            }
+            $this->setSwooleServer($server);
+        }
+        return $this->getSwooleServer();
+    }
```

Comment:
Please refactor this method to return the server instance early if already present, as this will put the bulk of the work as the body of the method in that situation:

```php
if ($this->swooleServer) {
    return $this->swooleServer;
}

$this->swooleServer = new SwooleHttpServer(/* ... */);
$options = array_replace($this->options, $appendOptions);
if ([] !== $options) {
    $this->swooleServer->set($options);
}
return $this->swooleServer;
```